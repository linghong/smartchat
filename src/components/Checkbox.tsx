import { FC, ChangeEvent, Dispatch, SetStateAction } from 'react';

interface CheckboxProps {
  label: string;
  setIsChecked: Dispatch<SetStateAction<boolean>>;
}

const Checkbox: FC<CheckboxProps> = ({ label, setIsChecked }) => {
  const handleCheckboxChange = (e: ChangeEvent<HTMLInputElement>) => {
    setIsChecked(e.target.checked);
  };

  return (
    <div className="my-2">
      <input
        type="checkbox"
        id="checkbox"
        className="rounded border-gray-300 text-blue-600 focus:border-blue-500 focus:ring focus:ring-blue-200 focus:ring-opacity-50"
        onChange={handleCheckboxChange}
      />
      <label htmlFor="checkbox" className="mx-3 text-gray-700">
        {label}
      </label>
    </div>
  );
};
export default Checkbox;
