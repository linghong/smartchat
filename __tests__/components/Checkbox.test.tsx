import { render, screen, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';

import Checkbox from '@/src/components/Checkbox';

describe('Checkbox Component', () => {
  let setIsChecked: jest.Mock;

  beforeEach(() => {
    setIsChecked = jest.fn();
    render(<Checkbox label="Check me" setIsChecked={setIsChecked} />);
  });

  afterEach(() => {
    setIsChecked.mockClear();
  });

  it('renders Checkbox component correctly', () => {
    expect(screen.getByText('Check me')).toBeInTheDocument();
    expect(screen.getByRole('checkbox')).not.toBeChecked();
  });

  it('handles checkbox change', () => {
    fireEvent.click(screen.getByRole('checkbox'));
    expect(setIsChecked).toHaveBeenCalledWith(true);
  });

  it('Checkbox component snapshot', () => {
    const { asFragment } = render(
      <Checkbox label="Check me" setIsChecked={setIsChecked} />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
