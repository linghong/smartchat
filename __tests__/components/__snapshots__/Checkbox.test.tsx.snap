// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox Component Checkbox component snapshot 1`] = `
<DocumentFragment>
  <div
    class="my-2"
  >
    <input
      class="rounded border-gray-300 text-blue-600 focus:border-blue-500 focus:ring focus:ring-blue-200 focus:ring-opacity-50"
      id="checkbox"
      type="checkbox"
    />
    <label
      class="mx-3 text-gray-700"
      for="checkbox"
    >
      Check me
    </label>
  </div>
</DocumentFragment>
`;
